### DISCUSSION TITLE
Conditions for checkboxes and dropdown (multiple) fields

### DISCUSSION THREAD

**Discussion by koksikus** [directus/directus#14033] [Archive]
I set myself one of the checkboxes and dropdown fields with several options to choose from. When I try to set a condition in another field depending on the selection in the indicated fields, I am not able to do it because it seems that the fields are filled with all the choices all the time. Changing the selection doesn't change anything (in conditions). The condition that works is "contains" but is always true, regardless of the choice. Other conditions are not met ... Am I doing something wrong?

**Comment by br41nslug** [comment]
Do you mind sharing an example of how you've configured the checkboxes/dropdowns and what conditions you've set?

**Reply by koksikus** [reply]
I did it this way (field name "Type"):
![2022-06-22 103503](https://user-images.githubusercontent.com/90528895/174984015-8305a16b-47ec-4030-9697-d174473391cc.jpg)

In the next field, I set a condition
![2022-06-22 103640](https://user-images.githubusercontent.com/90528895/174984174-ea53d686-ef8c-4eee-a309-4941d6539507.jpg)
![2022-06-22 103619](https://user-images.githubusercontent.com/90528895/174984211-6f2e1863-9cdd-4e0a-88b7-74c696ad1dbd.jpg)

The result is this. I can still edit the field.
![2022-06-22 103719](https://user-images.githubusercontent.com/90528895/174984432-20fad952-328f-41ce-b548-af49b0a914bf.jpg)

I am using Directus 9.12.1

**Reply by br41nslug** [reply]
As it is a JSON field you should use `contains` instead of `one of` as JSON fields are filtered as string. It is always coming back true because `Manufacturer` will never match `["Manufacturer",other options]`.

**Reply by koksikus** [reply]
It doesn't work with contains, and with contains insensitive it is always true ...
RAW value is:

```json
{
	"_and": [
		{
			"Type": {
				"_icontains": "Manufacturer"
			}
		}
	]
}
```

Result:
![2022-06-22 144107](https://user-images.githubusercontent.com/90528895/175031430-948169a0-2e63-4b1d-81a2-50ecb558c42c.jpg)
![2022-06-22 144036](https://user-images.githubusercontent.com/90528895/175031409-336cda66-4760-4cb8-aeed-340fc07e7fcf.jpg)

**Reply by br41nslug** [reply]
I did make the assumption that you've used the exact same `values` as the checkbox `texts` im seeing. Also question how are you selecting the text comparisons? in my experiments with 9.12.2 and 9.12.1 im only able to select `_null` and `_nnull` for my checkbox field 🤔

**Reply by koksikus** [reply]
If you use json then yes, you are right, conditions are just: _null and _nnull. But when creating the field I chose CSV and I can still use supposedly all conditions. The values and texts are the same.
Or maybe the condition checks only what results from the field settings, and not what is actually marked in this field.  If "Manufacturer" is in the settings, the condition is always met ...

**Reply by br41nslug** [reply]
sorry for the slow process but i've updated my test setup and can confirm that it's not working as intended (strange that JSON and CSV would behave differently in this situation but that might be a separate issue). The `one of` filter should work like you initially assumed but it is not. ~Could you open an issue for this?~ I think you might be running into this old (and still open) issue #11309

**Reply by benhaynes** [reply]
@br41nslug There's an option above to convert this Discussion into an issue if that's easier (and we keep all this info).

**Reply by br41nslug** [reply]
Thanks, i knew about `issue -> discussion` but it hadnt occurred to me to try the other way around 🤦

**Reply by rijkvanzanten** [reply]
"contains" currently is exclusively a string operation. We currently don't support "Array contains" yet. This is part of https://github.com/directus/directus/discussions/7277
